website/docs/cloud-docs/workspaces/dynamic-provider-credentials/hcp-configuration.mdx

```diff
@@ -0,0 +1,110 @@
+---
+page_title: Dynamic Credentials with the HCP Provider - Workspaces - HCP Terraform
+description: >-
+  Use OpenID Connect to get short-term credentials for the HCP Terraform provider in
+  your HCP Terraform runs.
+---
+
+# Dynamic Credentials with the HCP Provider
+
+~> **Important:** If you are self-hosting [Terraform Cloud agents](/terraform/cloud-docs/agents), ensure your agents use [v1.15.1](/terraform/cloud-docs/agents/changelog#1-15-1-05-01-2024) or above. To use the latest dynamic credentials features, [upgrade your agents to the latest version](/terraform/cloud-docs/agents/changelog).
+
+You can use HCP Terraform’s native OpenID Connect integration with HCP to get [dynamic credentials](/terraform/cloud-docs/workspaces/dynamic-provider-credentials) for the HCP provider in your HCP Terraform runs. Configuring the integration requires the following steps:
+
+1. **[Configure HCP](#configure-hcp):** Set up a trust configuration between HCP and HCP Terraform. Then, you must create an [service principal in HPC](https://developer.hashicorp.com/hcp/docs/hcp/admin/iam/service-principals) for your HCP Terraform workspaces.
+2. **[Configure HCP Terraform](#configure-hcp-terraform):** Add environment variables to the HCP Terraform workspaces where you want to use Dynamic Credentials.
+
+Once you complete the setup, HCP Terraform automatically authenticates to HCP during each run.
+
+## Configure HCP
+You must enable and configure a workload identity pool and provider on HCP. These instructions use the HCP CLI, but you can also use Terraform to configure HCP. Refer to our [example Terraform configuration](https://github.com/hashicorp/terraform-dynamic-credentials-setup-examples/tree/main/hcp).
+
+#### Create a service principal
+You will need to create a service principal for HCP Terraform to assume during runs. Remember the ID of the service principle as you will need it for the next steps.
```

Comment:
Suggestion: Should be "principal" instead of "principle"?:
```suggestion
You will need to create a service principal for HCP Terraform to assume during runs. Remember the ID of the service principal as you will need it for the next steps.
```